### ISSUE TITLE
mapRequestToAsset give 404`s on Gatsby projects

### ISSUE THREAD

**Issue by cristianCeamatuAssist** [cloudflare/kv-asset-handler#172]
Hello,

I am working on a Gatsby project that is being deployed to Cloudflare, Gatsby bundles page URL's like `enterprise.html` and `contact-us.html` to `/enterprise.html/index.html` and `contact-us.html/index.html`. When I deployed to Cloudflare I was receiving 404's as the worker was not finding the assets. I investigated further and found out that the reason is that the workers use the `mapRequestToAsset` function with the below cases:
```javascript
  if (pathname.endsWith('/')) {
    // If path looks like a directory append index.html
    // e.g. If path is /about/ -> /about/index.html
    pathname = pathname.concat('index.html')
  } else if (!mime.getType(pathname)) {
    // If path doesn't look like valid content
    //  e.g. /about.me ->  /about.me/index.html
    pathname = pathname.concat('/index.html')
  }
```
The problem here is that the worker receives a request like `example.com/enterprise.html` which does not pass the `} else if (!mime.getType(pathname)) {` test case and it does not get the `/index.html` added so the worker cannot find the resource. Is there any way I can tell `mapRequestToAsset` to mandatory add the `/index.html` ?

I was thinking of something similar:
```javascript
const mapRequestToAsset = (request: Request, forceAdd: string | undefined) => {
  const parsedUrl = new URL(request.url)
  let pathname = parsedUrl.pathname

  if (pathname.endsWith('/')) {
    // If path looks like a directory append index.html
    // e.g. If path is /about/ -> /about/index.html
    pathname = pathname.concat('index.html')
  } else if (!mime.getType(pathname)) {
    // If path doesn't look like valid content
    //  e.g. /about.me ->  /about.me/index.html
    pathname = pathname.concat('/index.html')
  } else if (forceAdd) {
    pathname = pathname.concat(forceAdd)
  }

  parsedUrl.pathname = pathname
  return new Request(parsedUrl.toString(), request)
}
```

I need to keep the `.html` pages and already asked Gatsby if they can change the way their bundle works but I don't think that is possible. I can create a PR if you are open to this idea or have any other suggestions.

**Comment by Cherry** [comment]
Once https://github.com/cloudflare/kv-asset-handler/pull/159 is merged, you can override the `mapRequestToAsset` function in its entirety, which would support this use-case.

**Comment by cristianCeamatuAssist** [comment]
Thanks for the reply @Cherry, I hope it gets merged soon. Have a great day!

**Comment by signalnerve** [comment]
Closed by #159
